### PR TITLE
Update RequesterAuthorizerWithErc721

### PR DESCRIPTION
**Pull request by bbenligiray**
Closes https://github.com/api3dao/airnode/issues/1675

@api3/airnode-protocol-v1@2.4.0 exports the audited RequesterAuthorizerWithErc721 and its addresses. This PR uses these instead of the contracts from `dev/`.

One question to @dcroote , is it a problem for RequesterAuthorizerWithErc721 to only have been deployed on ethereum and ethereum-goerli-testnet regarding https://github.com/api3dao/airnode/issues/1717 ?
Also, should https://github.com/api3dao/airnode/pull/1755 also handle  RequesterAuthorizerWithErc721?

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -65,4 +65,11 @@ module.exports = {
     'require-await': 'error',
     'no-return-await': 'error',
   },
+  settings: {
+    'import/resolver': {
+      node: {
+        paths: ['src'],
+      },
+    },
+  },
```
Comment (bbenligiray) [inline .eslintrc.js:74]: I had to apply https://stackoverflow.com/a/42498220 even though I'm not confident what it really does

File: packages/airnode-protocol/contracts/dev/RequesterAuthorizerWithErc721.sol
Comment (dcroote) [inline packages/airnode-protocol/contracts/dev/RequesterAuthorizerWithErc721.sol:1]: With both this and the `MockErc721.sol` contract deleted, we should be able to remove the hardhat overrides too:

https://github.com/api3dao/airnode/blob/fa3593b7a7e413cee588e59c9df3afda182f9242/packages/airnode-protocol/hardhat.config.js#L58-L77

File: .eslintrc.js (hunk 1 above)
Comment (dcroote) [reply .eslintrc.js:74]: Not entirely sure either, but I know I also fought with ESLint import errors previously in related work so glad there is something that appears to have worked.

**Review by dcroote — APPROVED** [review]
Nice- 1 response and 1 minor comment.

As for the other questions:

> is it a problem for RequesterAuthorizerWithErc721 to only have been deployed on ethereum and ethereum-goerli-testnet

Not a problem per se - we'll default to these addresses if the contract address is omitted on these chains. Otherwise, users will have to (deploy and) specify it themselves.

> Also, should https://github.com/api3dao/airnode/pull/1755 also handle RequesterAuthorizerWithErc721?

Yes- this PR is perfectly timed as I can fix the TODOs in that PR once this is merged

File: packages/airnode-protocol/contracts/dev/RequesterAuthorizerWithErc721.sol
Comment (bbenligiray) [reply packages/airnode-protocol/contracts/dev/RequesterAuthorizerWithErc721.sol:1]: Thanks, I wasn't aware of this

**Pull request merged**